pac4j-saml/src/main/java/org/pac4j/saml/crypto/KeyStoreCredentialProvider.java

```diff
@@ -1,145 +1,151 @@
-                } catch (final IOException e) {
-                    this.logger.debug("Error closing input stream of keystore", e);
-                }
-            }
-        }
-    }
-
-    private String getPrivateKeyAlias(final KeyStore keyStore) {
-        try {
-            final Enumeration<String> aliases = keyStore.aliases();
-            if (aliases.hasMoreElements()) {
-                return aliases.nextElement();
-            }
-
-            throw new SAMLException("Keystore has no private keys");
-
-        } catch (final KeyStoreException e) {
-            throw new SAMLException("Unable to get aliases from keyStore", e);
-        }
-    }
-}
+/*
+  Copyright 2012 - 2015 pac4j organization
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.pac4j.saml.crypto;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.opensaml.core.criterion.EntityIdCriterion;
+import org.opensaml.security.credential.Credential;
+import org.opensaml.security.credential.CredentialResolver;
+import org.opensaml.security.credential.impl.KeyStoreCredentialResolver;
+import org.opensaml.security.x509.X509Credential;
+import org.opensaml.xmlsec.config.DefaultSecurityConfigurationBootstrap;
+import org.opensaml.xmlsec.keyinfo.KeyInfoCredentialResolver;
+import org.opensaml.xmlsec.keyinfo.KeyInfoGenerator;
+import org.opensaml.xmlsec.keyinfo.NamedKeyInfoGeneratorManager;
+import org.opensaml.xmlsec.signature.KeyInfo;
+import org.pac4j.saml.exceptions.SAMLException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.shibboleth.utilities.java.support.resolver.CriteriaSet;
+import net.shibboleth.utilities.java.support.resolver.ResolverException;
+
+
+/**
+ * Base for all key store credential providers.
+ * 
+ * Responsible for loading a private key from a JKS keystore and returning the corresponding {@link Credential}
+ * opensaml object.
+ *
+ * It can be used as-is; or a subclass can be used instead, providing different constructor arguments for convenience.
+ * It is up to subclasses to provide an adequate input stream to the constructor, which should contain data of the JKS.
+ * 
+ * @author Misagh Moayyed
+ * @since 1.8.0
+ */
+public abstract class KeyStoreCredentialProvider implements CredentialProvider {

```

Comment:
Changes in commit 033ee5303c0ea7c0f5c0e5445bff1e4c670f23a7:
The class is now abstract and has 2 sub-classes. The constructor has been changed and little JavaDoc has been added.